### PR TITLE
Rest methods implemented

### PR DESCRIPTION
**Pull request by SirNaga**
-added Vehicle service
-added creating user to db
-added login for db user
-added creating rental
-added reading cars from db

### CODE REVIEW

File: CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java [hunk 1]
```diff
@@ -16,6 +20,24 @@
 @RequestMapping("/vehicle")
 public class CarController extends IController {
 
+    @Autowired
+    VehicleService vehicleService;
+
+    @GetMapping("/allVehicles")
```
Comment (omid-nazifi) [inline CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java]: benenn bitte das mit "all" um und entfern die untere Mock Methode
=> public ResponseEntity<List<Vehicle>> getAllCustomers() {

File: CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java [hunk 2]
```diff
@@ -16,6 +20,24 @@
...
+        return new ResponseEntity<>(vehicleService.getAllCustomerRentalForCustomer(customerId), HttpStatus.OK);
+    }
+
+    @PutMapping("/vehicleRent")
+    public ResponseEntity<CustomerRental> rentAVehicle(CustomerRental customerRental){
+        vehicleService.createVehicleRental(customerRental);
+        return new ResponseEntity<>(customerRental, HttpStatus.OK);
+    }
+
     @GetMapping(value = "/all")
```
Comment (omid-nazifi) [inline CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java]: das kann entfernen

File: CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java (hunk 1 above)
Comment (SirNaga) [reply CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java]: changed

File: CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java (hunk 2 above)
Comment (SirNaga) [reply CarRental/src/main/java/at/wiencampus/se/carrental/controller/CarController.java]: changed

**Review by omid-nazifi — APPROVED** [review]

**Pull request merged**